language-plutus-core/src/Language/PlutusCore/TypeSynthesis.hs

```diff
@@ -113,13 +118,12 @@ kindCheck :: (MonadError (Error a) m, MonadQuote m) => Natural -> Type TyNameWit
 kindCheck n t = convertErrors asError $ runTypeCheckM n (kindOf t)
 
 -- | Run the type checker with a default context.
-runTypeCheckM :: (MonadError (TypeError a) m, MonadQuote m)
-              => Natural -- ^ Amount of gas to provide typechecker
+runTypeCheckM :: Natural -- ^ Amount of gas to provide typechecker
               -> TypeCheckM a b
-              -> m b
+              -> ExceptT (TypeError a) Quote b
```

Comment:
I do not have any particular preference here, but I'm curious why you changes the signature.

Reply:
This was just what worked. If we need to modify it, that can be done.

Reply:
I don't see what was wrong with the old type signature either.

Reply:
Nothing is *wrong* per se, however it will not compile as currently written with the old type signature.

Reply:
Well you also changed the body...